### PR TITLE
Feature/lint+flow components

### PR DESCRIPTION
**Pull request by albertoestarrona**
> Please check you are not duplicating an existing pull request...

#### Changes

This PR includes the ESLint+Flow refactor for components:
- NationActionButton.
- PanelViewAlert.
- PanelViewCitizen.
- SwitchLabeled.

### CODE REVIEW

**Comment by seland** [comment]
@albertoestarrona Please check tests failed on semaphore CI.

File: src/components/common/NationActionButton.js [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
+  /**
+   * @desc Text label of the button.
+   */
+  title?: string,
+  /**
+   * @desc Callback to be called when button is pressed.
+   */
+  onPress?: () => void,
+  /**
+   * @desc Callback to be called on Switch when changes value.
```
Comment (seland) [inline src/components/common/NationActionButton.js]: Description seems to be not from here.

File: src/components/common/PanelViewAlert.js [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 import { MediaQueryStyleSheet } from 'react-native-responsive';
-import PropTypes from 'prop-types';
 import GlobalStyles from '../../global/Styles';
 import i18n from '../../global/i18n';
 
-const PanelViewAlert = ({ status }) => {
+
+type Props = {
+  /**
+   * @desc Statue message to show in panel
```
Comment (seland) [inline src/components/common/PanelViewAlert.js]: A typo 'Statue' -> 'Status' 🙂

File: src/components/common/NationActionButton.js [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
+   */
+  iconSource?: number,
+}
+
+/**
+ * @desc Component that renders the panel indicating the user is Citizen of a Nation
```
Comment (seland) [inline src/components/common/NationActionButton.js]: Description seems to be not from here.

File: src/components/common/NationActionButton.js (hunk 1 above)
Comment (albertoestarrona) [reply src/components/common/NationActionButton.js]: Fixed

File: src/components/common/PanelViewAlert.js (hunk 2 above)
Comment (albertoestarrona) [reply src/components/common/PanelViewAlert.js]: Fixed

File: src/components/common/NationActionButton.js (hunk 3 above)
Comment (albertoestarrona) [reply src/components/common/NationActionButton.js]: Fixed

**Pull request merged**